### DISCUSSION TITLE
Why do we need the `argoexec wait` container? Any special consideration?

### DISCUSSION THREAD

**Discussion by lotabout** [argoproj/argo-workflows#8918] [Q&A] [answered]
With the latest `emssiary` executor, user's commands are invoked by `argoexec emissary` in the main container. Currently
- `emissary` would collect the logs and artifacts of user's commands and save it to some directory under `/var/run/argo/...`
- The wait container(`argoexec wait`) would detect it and upload the logs and artifacts to artifact repositories.

My question is: why not handle the uploads in the `emissary` command? All I can think of is the limitation of evolution path where the old docker executor would submit user's commands and `wait` container was needed as `post-container`. Is it still valid now?

**Comment by lotabout** [comment, marked as answer]
Hi @alexec , could you please have a look at the above design question on argo emissary? I found that you wrote the code and maybe only the author could answer such questions :) Thanks!

**Reply by alexec** [reply]
This is the issue you are looking for https://github.com/argoproj/argo-workflows/issues/6902

**Reply by lotabout** [reply]
Thanks! Seems that I used the wrong keywords for searching. I'll checkout the issues.
